.github/workflows/main_ci.yml

```diff
@@ -6,53 +6,34 @@ on:
       - master
   pull_request:
 
-permissions:
-  contents: read
-
 jobs:
-  ##################
-  # Jobs with matrix
-  ##################
   unit:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        node-version: [14, 'lts/*']
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
     steps:
-      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@eeb10cff27034e7acf239c5d29f62154018672fd # v3
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v4
         with:
-          node-version: ${{ matrix.node-version }}
+          node-version: 18
           registry-url: https://registry.npmjs.org/
-          cache: 'npm'
```

Comment:
Should keep the matrix and just change the 14 to 18. `lts/*` will run the latest LTS (so currently v22)

Also, all the security stuff (read permissions etc.) should remain the same.

Reply:
Referring to tags by their commit is overkill tbh and it's find to use tags.